### DISCUSSION TITLE
Field On Change Save Action vs Button Save Action

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#10351] [Help] [answered]
I have implemented a Form inside a Repeater in the hopes of easily updating child rows under a parent detail. As part of the onchange event I validate the form, then update a field value, and then save the row during which I update two columns.

For the onchange actions for Meter_Reading, the form field Amount_Pumped gets correctly updated. Then for the save row action the previous values for Meter_Reading and Amount_Pumped are persisted however and therefore the values under Gallons and Acre-Feet are also not updated during the save row action. When saving the row via button click instead, all values are persisted correctly. I included a screenshot of the form field value along with the post request payload from network traffic:

![image](https://user-images.githubusercontent.com/106758851/233136343-49a53ffe-2b7a-42d3-aca6-16459fb0c99f.png)

I don't know if the Post for the update happens before the field value update and if there is a race condition under these circumstances or if something else is a factor. If that is the case is there an action where I could insert a JS Timeout() function between update field value and save row? Also below is a screenshot of the Meter_Reading field onchange actions:

![image](https://user-images.githubusercontent.com/106758851/233137469-9437523f-7bfa-49a4-9136-40aa4b3b79d1.png)

**Comment by andz-bb** [comment, marked as answer]
hey @mmalessa80, if you're using field onChange actions the value of the Form Fields binding will be the field's original value rather than the updated one. If you need to access the new value for the field, you can use the `{{ Field Value }}` binding instead. In this case if your onChange actions are on the `Meter_Reading` field, you can replace references to `{{ New Form.Fields.Meter_Reading }}` with `{{ Field Value }}` and it should fix it saving the old value.

**Reply by mmalessa80** [reply]
Thanks for the heads up, I figured I was missing something. My solution needed a bit more tweaking since the Meter_Reading updates the Amount_Pumped, and then there are two other columns that get updated based on the now updated Amount_Pumped. So in the end I used App State to calculate the amount that goes into Amount_Pumped and then used that state value to calculate the two other columns.
